packages/server/modules/workspaces/services/workspaceJoinRequests.ts

```diff
@@ -38,23 +47,38 @@ export const requestToJoinWorkspaceFactory =
     createWorkspaceJoinRequest,
     sendWorkspaceJoinRequestReceivedEmail,
     getUserById,
-    getWorkspace
+    getWorkspaceWithDomains,
+    getUserEmails
   }: {
     createWorkspaceJoinRequest: CreateWorkspaceJoinRequest
     sendWorkspaceJoinRequestReceivedEmail: SendWorkspaceJoinRequestReceivedEmail
     getUserById: GetUser
-    getWorkspace: GetWorkspace
+    getWorkspaceWithDomains: GetWorkspaceWithDomains
+    getUserEmails: FindEmailsByUserId
   }) =>
   async ({ userId, workspaceId }: { userId: string; workspaceId: string }) => {
     const requester = await getUserById(userId)
     if (!requester) {
       throw new NotFoundError('User not found')
     }
 
-    const workspace = await getWorkspace({ workspaceId })
+    const workspace = await getWorkspaceWithDomains({ id: workspaceId })
     if (!workspace) {
       throw new WorkspaceNotFoundError('Workspace not found')
     }
+    if (!workspace?.discoverabilityEnabled) throw new WorkspaceNotDiscoverableError()
+    const workspaceDomains = workspace.domains.filter((domain) => domain.verified)
+    if (!workspaceDomains.length) throw new WorkspaceNotJoinableError()
+
+    const userEmails = await getUserEmails({ userId })
```

Comment:
isn't there a logic function to check if the user is compliant with the workspace domain policies?